server/src/main/java/com/kentaromatt/todoapi/ToDoController.java

```diff
@@ -32,6 +35,28 @@
         }};
     }
 
+    @PostMapping(path = "/todos")
+    public @ResponseBody HashMap<String, String> addToDo(@RequestBody HashMap<String, String> requestMap) {
+        if (!requestMap.containsKey("description") || requestMap.get("description") == "") {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Description is required");
+        }
+        String description = requestMap.get("description");
```

Comment:
what happens if `requestMap.get("description")` returns not a `String`?

Reply:
Two considerations here:
- We're always sending a string representation of a json object from the frontend (as with any HTTP request)
- The controller mapping is expecting a `HashMap<String, String>` so I think we can also lean on that type checking?

Reply:
yep it'll return an error, so it should be fine